include/xtensor-python/pyarray.hpp

```diff
@@ -82,6 +82,12 @@ namespace xt
     };
 
     template <class T>
+    struct xiterable_inner_types<pyarray<T>>
+        : xcontainer_iterable_types<pyarray<T>>
+    {
```

Comment:
I think that you should override ```broadcast_iterator`` and ``const_broadcast_iterator`` types here since they use ``shape_type`` in ``xcontainer_iterable_types`` definition while they should use ``inner_shape_type`` for ``pyarray`` and ``pytensor``.